### ISSUE TITLE
Game logs over 300+Gigs

### ISSUE THREAD

**Issue by 001pickle** [Yellow-Dog-Man/Resonite-Issues#602]
### Describe the bug?

I'm not sure what caused this but after a while, my PC notified me that my drive was full, after some digging, it turns out Resonite has created log files 367gigs in size.

### To Reproduce

Run the game.

### Expected behavior

not create 300gigs of data

### Screenshots

Images were taken with my phone due to the system being so full. 
![IMG_1266](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/74747214/8a5aad16-2abf-4a48-a92a-9c67a5ef23b2)
![IMG_1267](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/74747214/06cccd70-8ac5-4a2c-8cbc-750322a79977)
![IMG_1265](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/74747214/fa1df6c5-8f29-476c-8570-0a2d8bac1523)

### Resonite Version Number

2023.11.8.335

### What Platforms does this occur on?

Windows

### What headset if any do you use?

Rifts/Pimax 5k Super

### Log Files

I can't load the log file as the system it's on has very slow internet, it would take 30 days. I have no way to move that large of a file to my main PC. (error was on my living room system)

### Additional Context

system is a 

5900X
64gigs ram
1080ti
1tb nvme

### Reporters

endurofurry - discord

**Comment by Stellanora64** [comment]
Is this consistent or have you noticed it only generates logs that big in a specific session, world, item etc?

**Comment by epicEaston197** [comment]
That is absolutely absurd
i have many many many more logs and they're all just normal I'm on the game for hours at a time
![image](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/76523692/fa927379-cfe4-469c-b92f-e3770fa66563)

**Comment by 001pickle** [comment]
I only have 4 sessions and have yet to leave the home world. I have been battling issues with my headset and steam so I have yet to even join other worlds or import my old neos stuff yet

**Comment by ShrikeAlvaron** [comment]
Those logs are never purged, you'll need to delete them.  Before you do though, could you see if you can open them and look at what's actually in there?  With files that big even opening them may be a problem though.

**Comment by 001pickle** [comment]
yea even note pad ++ won't open anything over 2gigs. I don't think the file is any use but I'm holding onto it until I hear back from a dev. regardless of if they auto delete or not it shouldn't be making logs 367gigs something is bugged

**Comment by ItsAlphaNeon** [comment]
That is actually insane, this should never happen in any case. Have you been able to find a way to inspect the log without opening it completely?

**Comment by Meister1593** [comment]
You probably need to cut some part of the log with other tool that doesn't read entire file at once.
Some suggest using powershell: `Get-Content yourfile.log -TotalCount 100 | Out-File -Path C:\out\path.txt`
 (it's either 100 lines, or 100k lines)

**Comment by 001pickle** [comment]
I have not found any way to open it without crashing the system or the tool to open it. I figured its now a waiting game to hear from the Devs to get their input as this is pretty out of my wheelhouse. unless its server based software or hardware I'm pretty uneducated past that. I'm going to leave it like is. and wait to hear back.

**Comment by Frooxius** [comment]
I would need to get a sample of these logs to see what is being spammed into them, otherwise it's hard to figure out what has caused this.

**Comment by TisFoolish** [comment]
Since the log files are in a txt format you could split the file. @001pickle, are you comfortable with running a power shell script? I or someone else could try their hand at making a powershell script to split off the first 10 megabytes.

**Comment by 001pickle** [comment]
@TisFoolish Yeah I'm not versed in power shell. If you could point me in the right direction I would appreciate it.

**Comment by Frooxius** [comment]
You could also try to compress one of the big logs and share that. They might compress to a tiny fraction of the original size.

**Comment by 001pickle** [comment]
Are you not concerned trying to compress it could cause data loss on such a large file? not only that, even if I could make it smaller It would still have to be split to look at it I presume. it would be passing the torch to someone else. 

I have been super busy the last couple of days but if no one has a solution I will look for one in the next few days to cut part of the file off.

**Comment by Frooxius** [comment]
@001pickle No, that's not how the compression works. ZIP, 7zip and so on are lossless compression formats, which mean that they do not lose any data.

It's easy for me to process big log file, so if you're having issues cutting a part of the log file off, this might make it easier, as I'll just do it on my end.

**Comment by TisFoolish** [comment]
@001pickle Here is a simple powershell script.  It'll split of the first 5MB of any txt file in the same folder as it and put them in a folder caled "done".

Instructions:

Copy and paste it to a text file. 

Save that text file to a new, empty folder. 

**CHANGE THE EXTENSION TO `.ps1`.**

Move at least one of the log files to that new folder.

Hold shift and right click inside the folder

Select "Open PowerShell Window Here"
![image](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/38511571/e85dc161-5940-493b-9ef8-4403bf1c2d49)

type `./` into the window and hit tab until the powershell script is selected

Hint enter



```powershell
$fileList = GCI -Path $PSScriptRoot | ?{$_.Extension -in ".txt"}
If(!(test-path -PathType container $PSScriptRoot\done\))
{
      New-Item -ItemType Directory -Path $PSScriptRoot\done\
}
ForEach ($f in $fileList)
{
	$from = $f.Fullname
	$outFile = "$($PSScriptRoot)\done\$($f.Basename)"
	$upperBound = 5MB


	$fromFile = [io.file]::OpenRead($from)
	$buff = new-object byte[] $upperBound
	$count
	"Reading $upperBound"
	$count = $fromFile.Read($buff, 0, $buff.Length)
	$to = "{0} split.txt" -f ($outFile)
	$toFile = [io.file]::OpenWrite($to)

	$tofile.Write($buff, 0, $count)

	$tofile.Close()
	
}
$fromFile.Close()


```

**Comment by 001pickle** [comment]
@Frooxius oh! I thought you were talking about something else. I didn't think Zip would have the power to make that significantly small enough. I zipped it here you go. https://drive.google.com/file/d/1iBvkTP1Q4wQEzOvsL9nwOM-S2c2s91wH/view?usp=sharing 

PLEASE BE ADVISED THIS IS KINDA A ZIP BOMB DUE TO IT CONTAINING A SINGLE 300+GB TXT.

**Comment by StormyDragon** [comment]
[looping.log](https://github.com/Yellow-Dog-Man/Resonite-Issues/files/13327649/looping.log)

Attached the first 200k bytes of the log.
It seems predictable after that; endless `System.OutOfMemoryException`

**Comment by 001pickle** [comment]
System has 64gigs of ddr4 4200 CL16 for memory. shouldnt be having this issue. It was running fine while generating these errors so I would rule out a ram leak. This system also hosts a MC modded server which i got no complaints from any of the players. I cross checked the system log from the MC server and it said 23.4gigs of ram were used total on the system at the time of these this log. So what now?

**Comment by Frooxius** [comment]
Yes, that's sufficient. 

There is huge amount of OutOfMemoryExceptions in the log. There's not really much we can do with this, as it's an internal error from the runtime. If that happens, the only choice we really have is crash the whole app, so it stops spewing errors.

@001pickle Do you have swap (virtual memory) disabled? What is the memory usage? Do you have anything that would restrict how much memory can be allocated to Resonite?

**Comment by 001pickle** [comment]
@Frooxius It's a default Windows 10 install. the only thing that's changed is Defender is disabled, sleep is disabled, and power settings are on performance. I had not had time to make the optimizations to the system yet to make it more efficient, Thread Scheduling changes, etc. Memory was at 36% used. its 64gigs of DDR4 CL16, 4200hz. the system was running fine. the only reason I noticed something was wrong is it let me know one of my drives was nearly full. that's when I closed resonite and looked into it.. After you mentioned above that it was throwing out of memory errors I considered it a potential RAM leak. but I doubt that because of how much it says was used. Not only that the system seemed completely happy to navigate and run while the game was on before I closed it. No studders, nothing. This system has been running for 3-4 years with no flaws 24hours a day and still has not had any issues since I stopped playing resonite on it. I'm honestly super interested in what could cause such an error set. I have never seen this kinda error ever in many many years of gaming.

**Comment by Frooxius** [comment]
Did this only happen once? It's hard to say what could've caused this from just this. 

The only thing here is to crash whole client when out of memory exception occurs, we can't really recover from that.

**Comment by 001pickle** [comment]
I never ran resonite on the system again. I could run some tests and see if it would reproduce. If you would like that. please let me know all the data you want me to get while the game is running and it's doing this.  

Maybe a warning that yells at a user an error has happened and to please restart their game? Have something looking for error logs over a specific size? My only request would be to get a warning so I could mitigate this issue before the drive gets so full Windows blue screens and doesn't recover because every bit on the drive is full.

**Comment by Frooxius** [comment]
Unfortunately warning doesn't look like it would be possible, as that would require allocating new items and objects in memory - but that's precisely what's failing.  Like I mentioned in my post above, the only thing we can reasonably do at that point is force the app shut down, because it cannot continue operating normally. 

If you have any steps that would help reproduce this, that would help. Any behavior or any details you notice when the bug occurs would help also.

**Comment by 001pickle** [comment]
I will get onto attempting to reproduce the failure within the following months. I'm kinda busy so please leave this thread up and I will report back with findings. If I find it actively creating these logs what data would you want me to get while it does this?

**Comment by Frooxius** [comment]
I've actually implemented mechanism to force crash in 2024.1.26.617, so the huge log issue should not happen anymore for this particular cause.

I'm closing the issue. But if you happen to replicate it, we can re-open this. Alternatively if you find a different, but related issue, you can open a new issue for that.

Thanks for the cooperation!
